Use editorConfig.theme for CodeMirror theme

davidjb opened the pull request:
This fixes #788 by firstly testing for `editorConfig.theme` in the config and using that if present.  This falls back to using `highlightTheme` (deprecated) so backwards compatibility is assured.

sapegin commented on scripts/make-webpack-config.js:
```diff
@@ -31,7 +31,9 @@ module.exports = function(config, env) {
 			extensions: ['.js', '.jsx', '.json'],
 			alias: {
-				'rsg-codemirror-theme.css': `codemirror/theme/${config.highlightTheme}.css`,
+				'rsg-codemirror-theme.css': `codemirror/theme/${(config.editorConfig &&
```
There's already a fallback in the config scheme, `highlightTheme` should only be used there.

davidjb replied:
Thanks for pointing that out, I've updated the PR accordingly to just use `editorConfig.theme` in that case.

davidjb commented:
@sapegin Thanks for pointing that out; I've removed it from  `styleguide-loader.js` and also changed the webpack alias to just use `editorConfig.theme` as per your suggestion.

sapegin commented on scripts/make-webpack-config.js:
```diff
@@ -31,7 +31,8 @@ module.exports = function(config, env) {
 			extensions: ['.js', '.jsx', '.json'],
 			alias: {
-				'rsg-codemirror-theme.css': `codemirror/theme/${config.highlightTheme}.css`,
+				'rsg-codemirror-theme.css': `codemirror/theme/${config.editorConfig &&
```
This condition doesn’t make sense:

1. `config.editorConfig` is always defined (check the schema)
2. If it wasn’t defined, you’d have a broken path.

davidjb replied:
Updated :+1:.

sapegin commented:
Thanks, merged!
